extensions-core/kafka-indexing-service/src/main/java/io/druid/indexing/kafka/supervisor/KafkaSupervisor.java

```diff
@@ -1630,8 +1635,14 @@ private void checkPendingCompletionTasks() throws ExecutionException, Interrupte
 
           // reset partitions offsets for this task group so that they will be re-read from metadata storage
           partitionGroups.get(groupId).replaceAll((partition, offset) -> NOT_SET);
-          sequenceTaskGroup.remove(generateSequenceName(groupId));
-
+          if (taskGroups.get(groupId) == null) {
+            log.warn(
+                "group id: %s not found in taskGroups, skipping remove from sequenceTaskGroup or it might have already been removed",
```

Comment:
What do you mean by 'skipping remove from sequenceTaskGroup _or_ it might have already been removed'?

Reply:
> which makes me think, if the above 2 operations should be atomic ?

I'm not sure. @dclim any thoughts?